test/unit/lib/httpUtilTest.js

```diff
@@ -0,0 +1,34 @@
+/* global describe, it */
+const httpUtil = require('../../../app/common/lib/httpUtil')
+const assert = require('assert')
+
+require('../braveUnit')
+
+describe('httpUtil test', function () {
+  describe('responseHasContent', function () {
+    describe('expected success codes', function () {
+      it('returns true for various success responses (200, 203, 206)', function () {
+        assert.equal(httpUtil.responseHasContent(200), true)
+        assert.equal(httpUtil.responseHasContent(203), true)
+        assert.equal(httpUtil.responseHasContent(206), true)
+      })
+      it('returns true for a cached response (304)', function () {
+        assert.equal(httpUtil.responseHasContent(304), true)
+      })
+    })
+
+    describe('expected failure codes', function () {
+      it('returns false for non-content success codes (used for REST apis, etc)', function () {
+        assert.equal(httpUtil.responseHasContent(201), false) // created
+        assert.equal(httpUtil.responseHasContent(202), false) // accepted
+      })
```

Comment:
4xx? Don't necessarily need an exhaustive list, but the list isn't that long https://en.wikipedia.org/wiki/List_of_HTTP_status_codes
can be done in a followup too